projects/sdk-wells/test/swap.test.ts

```diff
@@ -0,0 +1,198 @@
+import { Well } from "../src/lib/Well";
+import { ERC20Token, Token, TokenValue } from "@beanstalk/sdk-core";
+import { getTestUtils } from "./TestUtils/provider";
+import { deployTestWellInstance } from "./TestUtils";
+
+const { wellsSdk, utils, account } = getTestUtils();
+
+const DAI = new ERC20Token(
+  wellsSdk.chainId,
+  "0x6B175474E89094C44Da98b954EedeAC495271d0F",
+  6,
+  "DAI",
+  {
+    name: "DAI",
+    displayDecimals: 2
+  },
+  wellsSdk.provider
+);
+
+jest.setTimeout(30000);
+
+beforeAll(async () => {
+  await utils.resetFork();
+});
+
+const setupWell = async (wellTokens: ERC20Token[], account: string) => {
+  // Deploy test well
+  const deployment = await deployTestWellInstance(wellTokens);
+  const testWell = new Well(wellsSdk, deployment.wellAddress);
+
+  // Set initial balances for all well tokens
+  await Promise.all(
+    wellTokens.map(async (token) => {
+      await utils.setBalance(token, account, token.amount(30000));
+    })
+  );
+
+  await utils.mine();
+
+  // TODO: Why does this fail? While the original code below works?
+  // for (let i = 0; i++; i < wellTokens.length) {
+  //   await wellTokens[i].approve(testWell.address, TokenValue.MAX_UINT256.toBigNumber());
+  // }
+
+  // await wellTokens[0].approve(testWell.address, TokenValue.MAX_UINT256);
+  // await wellTokens[1].approve(testWell.address, TokenValue.MAX_UINT256);
+
+  // wellTokens.forEach(async (token) => {
+  //   await token.approve(testWell.address, TokenValue.MAX_UINT256);
+  // });
+
+  // Original code
+  await Promise.all([
+    await wellsSdk.tokens.BEAN.approve(testWell.address, TokenValue.MAX_UINT256.toBigNumber()),
+    await wellsSdk.tokens.WETH.approve(testWell.address, TokenValue.MAX_UINT256.toBigNumber())
```

Comment:
don't need to await here

Reply:
Are you sure? It returns a promise 🤔 And this is the same way it's done in the sdk swap tests.